### PR TITLE
A couple of optimizations and refinements

### PR DESCRIPTION
**Pull request by gramalingam**
Extract the independent optimization/refinements from [the fusion PR](https://github.com/microsoft/onnxscript/pull/1938) as a separate PR, ready to be reviewed/merged. (The fusion work is still WIP.)

* Replace Expand by Identity when applicable (in core optimization)
* Cleanup Dropout Identity replacement in the case when Dropout has mask output
* Make repeated (redundant) call to inliner efficient

### CODE REVIEW

File: onnxscript/optimizer/_constant_folding.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 import numpy as np
 import onnx
+import onnx.helper
```
Comment (justinchuby) [inline onnxscript/optimizer/_constant_folding.py:16]:
```suggestion
```

ditto

**Comment by justinchuby** [comment]
```
>>> ir.tensor([True])
Tensor<BOOL,[1]>(array([ True]), name=None)
```

This works for me?

also vscode doesn't seem to complain:

image

**Comment by gramalingam** [comment]
> ```
> >>> ir.tensor([True])
> Tensor<BOOL,[1]>(array([ True]), name=None)
> ```
> 
> This works for me?
> 
> also vscode doesn't seem to complain:
> 
> image

Oh, sorry. I had a typo, I did `ir.Tensor(...)` instead of `ir.tensor(...)` and that complained that an array-compatible object was required :-(

File: onnxscript/optimizer/_constant_folding.py [hunk 2]
```diff
@@ -448,25 +448,54 @@ def concat(node: ir.Node, op, state: OptimizerState) -> ReturnValue:
...
-        # TODO: handle this case. But unlikely to be needed in practice.
-        return None
+
+    def optimized_dropout():
+        input = node.inputs[0]
+        output = op.Identity(input)
+        if len(node.outputs) == 1:
+            return output
+        else:
+            true_tensor = ir.tensor([True])
```
Comment (justinchuby) [inline onnxscript/optimizer/_constant_folding.py:458]: Does it need to be 1d? Or is 0d better? No big deal though

**Review by justinchuby — APPROVED** [review]

**Pull request merged**